include/RAJA/DepGraphNode.hxx

```diff
@@ -84,43 +86,59 @@ public:
    /// algorithms and platforms. We haven't determined the best defaults yet!
    ///
    static const int _MaxDepTasks_          = 8;
-   static const int _SemaphoreValueAlign_  = 256;
 
    ///
    /// Default ctor initializes node to default state.
    ///
    DepGraphNode()
       : m_num_dep_tasks(0),
         m_semaphore_reload_value(0),
-        m_semaphore_value(0) 
-   { 
-      posix_memalign((void **)(&m_semaphore_value), 
-                     _SemaphoreValueAlign_, sizeof(int)) ;
-      *m_semaphore_value = 0 ;       
-   }
-
-   ///
-   /// Dependency graph node dtor.
-   ///
-   ~DepGraphNode() { if (m_semaphore_value) free(m_semaphore_value); }
+        m_semaphore_value(0) { }
 
    ///
    /// Get/set semaphore value; i.e., the current number of (unsatisfied)
    /// dependencies that must be satisfied before this task can execute.
    ///
-   int& semaphoreValue() {
-      return *m_semaphore_value ;
+   std::atomic<int>& semaphoreValue() {
+      return m_semaphore_value ;
    }
 
    ///
    /// Get/set semaphore "reload" value; i.e., the total number of external
    /// task dependencies that must be satisfied before this task can execute.
    ///
-   int& semaphoreReloadValue() {
+   std::atomic<int>& semaphoreReloadValue() {
       return m_semaphore_reload_value ;
    }
 
    ///
+   /// Ready this task to be used again
+   ///
+   void reset() {
+       m_semaphore_value.store(m_semaphore_reload_value);
+   }
+
+   ///
+   /// Satisfy one incoming dependency
+   ///
+   void satisfyOne() {
+       if (m_semaphore_value > 0) {
```

Comment:
It's mainly just protection against underflow to preserve the check against 0.  You're right that it shouldn't be required when used correctly, but I thought it would be safer to have the check.